src/main/kotlin/com/charliechauri/pokemon/api/query/PokemonQuery.kt

```diff
@@ -18,4 +21,10 @@ class PokemonQuery : Query {
 
         pokemonAdapter.getPokemonByName(name)
     }
-}
\ No newline at end of file
+
+    @GraphQLDescription("Ash's pokemons")
+    @GraphQLUnion(name = "AshPokemons", possibleTypes = [Charizard::class, Blastoise::class])
+    suspend fun getAshPokemon(): Any = coroutineScope {
```

Comment:
when changing `Any` to `BasicTypePokemon` there's a compilation error

```
2021-08-06 14:57:38.224  INFO 3516 --- [main] c.c.pokemon.api.ApplicationKt            : Starting ApplicationKt using Java 11.0.10

2021-08-06 14:57:38.229  INFO 3516 --- [main] c.c.pokemon.api.ApplicationKt            : No active profile set, falling back to default profiles: default
2021-08-06 14:57:41.330  WARN 3516 --- [main] onfigReactiveWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'webHandler' defined in class path resource [org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration$EnableWebFluxConfiguration.class]: Initialization of bean failed; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'routerFunctionMapping' defined in class path resource [org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration$EnableWebFluxConfiguration.class]: Invocation of init method failed; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.expediagroup.graphql.server.spring.GraphQLRoutesConfiguration': Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'springGraphQLServer' defined in class path resource [com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.class]: Unsatisfied dependency expressed through method 'springGraphQLServer' parameter 2; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'graphQLRequestHandler' defined in class path resource [com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.class]: Unsatisfied dependency expressed through method 'graphQLRequestHandler' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'graphQL' defined in class path resource [com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.class]: Unsatisfied dependency expressed through method 'graphQL' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'schema' defined in class path resource [com/expediagroup/graphql/server/spring/NonFederatedSchemaAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [graphql.schema.GraphQLSchema]: Factory method 'schema' threw exception; nested exception is com.expediagroup.graphql.generator.exceptions.InvalidCustomUnionException: The custom union annotation was used but the return type was com.charliechauri.pokemon.api.models.BasicTypePokemon instead of Any
2021-08-06 14:57:41.360  INFO 3516 --- [main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-08-06 14:57:41.405 ERROR 3516 --- [main] o.s.boot.SpringApplication               : Application run failed

```

Reply:
`The custom union annotation was used but the return type was com.charliechauri.pokemon.api.models.BasicTypePokemon instead of Any`